buffer/src/test/java/io/netty/buffer/PooledByteBufAllocatorTest.java

```diff
@@ -661,4 +661,53 @@ public void testCacheWorksForNormalAllocations() {
         assertEquals(3, allocator.metric().directArenas().get(0).numNormalAllocations());
         buffer.release();
     }
+
+    @Test
+    public void testNormalPoolSubpageRelease() {
+        // 16 < elemSize <= 7168 or 8192 < elemSize <= 28672, 1 < subpage.maxNumElems <= 256
+        // 7168 <= elemSize <= 8192, subpage.maxNumElems == 1
+        int elemSize = 32;
+        int oneLength = 16;
+        int fixedLength = 256;
+        ByteBuf[] ones = new ByteBuf[oneLength];
+        ByteBuf[] fixedByteBuf = new ByteBuf[fixedLength];
+        final PooledByteBufAllocator allocator = newAllocator(false);
+
+        for (int i = 0; i < fixedLength; i++) {
+            fixedByteBuf[i] = allocator.heapBuffer(elemSize, elemSize);
+        }
+
+        PoolSubpage<?>[] subpages = unwrapIfNeeded(fixedByteBuf[0]).chunk.subpages;
```

Comment:
Try to write a test that verifies on the externally observable behaviour of the allocator, rather than asserting on internal state changes.

Tests that rely on implementation internals tend to be more brittle, and easier to break on unrelated changes.

Reply:
the externally observable behavior of the allocator is when 'maxNumElems == 1' then release subpage operate not work, have no idea how to asserting that @chrisvest